### ISSUE TITLE
Support creating k8s secrets without the need to mount on to a pod

### ISSUE THREAD

**Issue by adusumillipraveen** [Azure/secrets-store-csi-driver-provider-azure#431]
**Describe the solution you'd like**

I am not sure if this is best place to raise this, but we thought it would be great to know your thoughts. 

Currently we use tools like https://github.com/bitnami-labs/sealed-secrets to manage secrets from azure keyvaults in gitops way. This repo supports a feature where a k8s secret can be created  using secretproviderclass and it can used to refer as environment variable in pod. Could the feature be extended in a way that we could create secrets even without mounting volumes onto pods? 

**Anything else you would like to add:**

If you are wondering why we cannot just add as a volume to the pod, in many cases we use helm charts/ deployment specs form third parties which doesn't support overriding volumemounts. so, it would be good to have a way to just create a k8s secret using this operator so that we could use those secrets.

**Environment:**

- Secrets Store CSI Driver version: (use the image tag): NA
- Azure Key Vault provider version: (use the image tag): NA
- Kubernetes version: (use `kubectl version`): NA
- Cluster type: (e.g. AKS, aks-engine, etc): NA

**Comment by mt35-rs** [comment]
I don't think this is a duplicate.  Azure is a commercial service with paying customers.  As such, it should listen to those customer use cases since Azure can always add functionality above and beyond the CSI driver spec.  The problem with the other issue tracker is that it is focused exclusively on CSI driver conformance and so comments like this are out of scope (except to propose modifications to the CSI driver spec...which is a longer term proposition).

As an Azure customer, I think this is a very important concern.  So many things rely on `Secret` objects and providers that mount secrets in `Pod`s just don't seem terribly useful to me.

**Comment by aramase** [comment]
Thank you for the feedback! 

I understand the concern. This provider is specifically implemented for the Secrets Store CSI Driver which relies on the CSI spec to invoke a provider, get secrets and write to file system. The provider implementation is only to interact with Key Vault, get the secrets and return back to the driver. What the driver does with the secret isn't something the provider is aware of. Having said that, I'm not sure if we can't make any changes in this provider to not rely on the mount/just create Kubernetes secret bypassing the driver. The entire model relies on being invoked by the `kubelet` process on the node to invoke the driver to perform an action.

Let me know what you think! I'm happy to engage and come up with something that's easier to configure and makes it a better experience compared to what we have today.

**Comment by mt35-rs** [comment]
As a user, I'm not so concerned about the CSI Driver.  What I'd like is the ability to synchronize secrets with Azure key vault.  There are existing tools out there, _e.g.,_ https://github.com/external-secrets/external-secrets, but I'd personally be a little more comfortable with something provided directly by Microsoft/Azure.  The reason being...secrets are a very sensitive thing.  While I generally like open source tools for the ability to audit them, I don't have time to audit them myself and I'm only interested in Azure Key Vault.  So an integrated capability in Azure provided by Microsoft would be a more "trustworthy" approach.  But what I need are first class `Secret`s.
